gobblin-api/src/main/java/gobblin/source/workunit/MultiWorkUnit.java

```diff
@@ -152,4 +152,14 @@ public int hashCode() {
   public static MultiWorkUnit createEmpty() {
     return new MultiWorkUnit();
   }
+
+  public void removeWorkUnits(WorkUnitStream workUnitsToRemove) {
+    for (WorkUnit workUnit : workUnitsToRemove.getMaterializedWorkUnitCollection()) {
```

Comment:
Please don't use `getMaterializedWorkUnitCollection`, it prevents optimizations in the Gobblin flow. Instead, you should use a lambda on the `WorkUnitStream`.